agent/internal/agent.go

```diff
@@ -52,7 +52,8 @@ type agent struct {
 	masterProto  string
 	masterClient *http.Client
 
-	reconnecting bool
+	reconnecting         bool
+	reconnectSendBacklog []interface{}
```

Comment:
I wish we could have a "reliability" wrapper around the sockets so all the backlog retry logic was hidden away in there, and not be an intrinsic part of `agent` actor here.

Reply:
that is my aim in a follow up pr, but this one needs to make release it seems